x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/analysis/index/IndexResolverTests.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.analysis.index;
+
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.xpack.sql.type.EsField;
+import org.elasticsearch.xpack.sql.type.TypesTests;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Map;
+
+public class IndexResolverTests extends ESTestCase {
+
+    @Test
+    public void testMergeSameMapping() throws Exception {
+        Map<String, EsField> oneMapping = TypesTests.loadMapping("mapping-basic.json", true);
+        Map<String, EsField> sameMapping = TypesTests.loadMapping("mapping-basic.json", true);
+        assertNotSame(oneMapping, sameMapping);
+        assertEquals(oneMapping, sameMapping);
+
+        String wildcard = "*";
+        IndexResolution resolution = IndexResolver.merge(
+                Arrays.asList(IndexResolution.valid(new EsIndex("a", oneMapping)), IndexResolution.valid(new EsIndex("b", sameMapping))),
+                wildcard);
+
+        assertTrue(resolution.isValid());
+
+        EsIndex esIndex = resolution.get();
+
+        assertEquals(wildcard, esIndex.name());
+        assertEquals(sameMapping, esIndex.mapping());
+    }
+
+    @Test
+    public void testMergeDifferentMapping() throws Exception {
+        Map<String, EsField> oneMapping = TypesTests.loadMapping("mapping-basic.json", true);
+        Map<String, EsField> sameMapping = TypesTests.loadMapping("mapping-basic.json", true);
+        Map<String, EsField> differentMapping = TypesTests.loadMapping("mapping-numeric.json", true);
+
+        assertNotSame(oneMapping, sameMapping);
+        assertEquals(oneMapping, sameMapping);
+        assertNotEquals(oneMapping, differentMapping);
+
+        String wildcard = "*";
+        IndexResolution resolution = IndexResolver.merge(
+                Arrays.asList(IndexResolution.valid(new EsIndex("a", oneMapping)),
+                        IndexResolution.valid(new EsIndex("b", sameMapping)),
+                        IndexResolution.valid(new EsIndex("diff", differentMapping))),
```

Comment:
Worth putting the different mapping as first in the List? `(EsIndex("diff"...), EsIndex("same"..), EsIndex("one"))`

Reply:
The order is somewhat important for this test since the comparison is done between the first entry and the rest.
The order above tests that a and b pass and that a and diff fail - there are two comparisons applied.
Putting diff first will cause only one comparison since and and b are not compared.
A minor detail that was done on purpose.